rfcs/RFC-Uploads.md

```diff
@@ -0,0 +1,348 @@
+
+**Authors:** [Trent Smith](mailto:trent.smith@chanzuckerberg.com)
+
+**Approvers:** [Friendly Tech Person](mailto:some.nice.person@chanzuckerberg.com), [Friendly PM Person](mailto:some.nice.person@chanzuckerberg.com), [Friendly Tech Person2](mailto:some.nice.person@chanzuckerberg.com), [Girish Patangay](mailto:girish.patangay@chanzuckerberg.com), [Sara Gerber](mailto:sara.gerber@chanzuckerberg.com)
+
+## tl;dr
+
+A user can upload files of any size to Data Portal using either a shared link from their cloud provider, or directly from their browser.
+
+## Glossary
+
+- CSP(Cloud Storage Provider) - Cloud storage services such as Dropbox or Google Drive.
+- Shared Link - A link from a CSP that give the bearer of that link access to private data.
+- DP(Data Portal) - The cellxgene Data Portal.
+- Small File - a file < 5 GB in size.
+- Large File - a file > 5 GB in size.
+
+## Problem Statement | Background
+
+One of the primary goals of the DP is to allow users to upload their own data sets. These data sets can vary in
+size from a few MB to several GB. User may want to upload from their local machine or from their CSP. For
+users that want to upload from their local machine, they must be able to resume an upload in the event their connection
+is interrupted. For users that want to upload data sets from their CSP, they are able to provide a shared link from
+their CSP that we can used to upload directly to our storage.
+
+## Product Requirements
+
+1. A user can upload a file to DP using a shared link from their CSP.
+1. A user can upload a file from their local machine using the DP browser.
+1. A user can resume an upload with out completely restarting.
+1. The user must be authenticated before uploading.
+
+### Nonfunctional requirements
+
+Nonfunctional requirements are not directly related to the user story but may instead reflect some technical aspect of the implementation, such as latency. [Here's](https://en.wikipedia.org/wiki/Non-functional_requirement) a more comprehensive list of nonfunctional requirements.
+
+1. Cloud to Cloud uploads will retry if they fail.
+1. A user can upload a single file up to 30GB in size.
+
+## Detailed Design | Architecture | Implementation
+
+![Upload Design](https://app.lucidchart.com/publicSegments/view/a74f747f-d071-41ea-949f-52f38a98747a/image.png)
+Figure 1: The Upload Design
+
+### Data Portal Browser App
+
+The DP Browser App will be used authenticate the user, and interact with the DP Backend to upload files.
+It will also use AWS JS SDK to upload files from local storage to AWS. These feature will extend the existing DP
+Frontend.
+
+### Data Portal Backend
+
+The DP Backend is responsible for handling users requests. It will generated presigned URL and access tokens
+for the DP Browser App to facilitate the uploading of files. It will also enqueue shared links for downloading.
+This additional functionality can be built into the existing DP Backend Lambda.
+
+### Download Queue
```

Comment:
Why are there references to _Download_ in an RFC about _Upload_?

Reply:
because this queue it to download the files from the provided link, and then upload to our s3 storage.

Reply:
I added more description